aiida_mlip/calculations/geomopt.py

```diff
@@ -0,0 +1,114 @@
+"""Class to run geom opt calculations."""
+
+from aiida.common import datastructures
+import aiida.common.folders
+from aiida.engine import CalcJobProcessSpec
+import aiida.engine.processes
+from aiida.orm import Bool, Float, SinglefileData, Str, StructureData, TrajectoryData
+
+from aiida_mlip.calculations.singlepoint import Singlepoint
+
+
+class GeomOpt(Singlepoint):  # numpydoc ignore=PR01
+    """
+    Calcjob implementation to run geometry optimization calculations using mlips.
+
+    Methods
+    -------
+    define(spec: CalcJobProcessSpec) -> None:
+        Define the process specification, its inputs, outputs and exit codes.
+    prepare_for_submission(folder: Folder) -> CalcInfo:
+        Create the input files for the `CalcJob`.
+    """
+
+    _DEFAULT_TRAJ_FILE = "aiida-traj.xyz"
+
+    @classmethod
+    def define(cls, spec: CalcJobProcessSpec) -> None:
+        """
+        Define the process specification, its inputs, outputs and exit codes.
+
+        Parameters
+        ----------
+        spec : `aiida.engine.CalcJobProcessSpec`
+            The calculation job process spec to define.
+        """
+        super().define(spec)
+
+        # Additional inputs for geometry optimization
```

Comment:
Depending on what we merge first, this will probably also need a --steps option, just as a heads up.

Reply:
my preference is to have the steps in...

Reply:
so I add it now and we mergen janus firt and then this

Reply:
yap

Reply:
steps are in and also --traj